pkg/scheduler/plugins/proportion/proportion.go

```diff
@@ -235,18 +235,22 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 		return victims, util.Permit
 	})
 
-	ssn.AddOverusedFn(pp.Name(), func(obj interface{}) bool {
-		queue := obj.(*api.QueueInfo)
+	ssn.AddOverusedFn(pp.Name(), func(queue *api.QueueInfo) (bool, bool) {
 		attr := pp.queueOpts[queue.UID]
+		isOverUsed := !attr.allocated.LessEqual(attr.deserved, api.Zero)
+		isAllDimensionOverUsed := attr.deserved.LessEqual(attr.allocated, api.Zero)
+		klog.V(3).Infof("Queue <%v>: deserved <%v>, allocated <%v>, share <%v>",
+			queue.Name, attr.deserved, attr.allocated, attr.share)
+		metrics.UpdateQueueOverused(attr.name, isOverUsed)
 
-		overused := !attr.allocated.LessEqual(attr.deserved, api.Zero)
-		metrics.UpdateQueueOverused(attr.name, overused)
-		if overused {
-			klog.V(3).Infof("Queue <%v>: deserved <%v>, allocated <%v>, share <%v>",
-				queue.Name, attr.deserved, attr.allocated, attr.share)
-		}
+		return isOverUsed, isAllDimensionOverUsed
+	})
+
+	ssn.AddOverusedForTask(pp.Name(), func(taskInfo *api.TaskInfo, queueID api.QueueID) bool {
+		podRequest := taskInfo.Resreq.Clone()
+		queueAllocated := pp.queueOpts[queueID].allocated.Clone()
 
-		return overused
+		return !podRequest.AddBaseLeft(queueAllocated).LessEqual(pp.queueOpts[queueID].deserved, api.Zero)
```

Comment:
Why need to add "AddBaseLeft" func here?

Reply:
if the GPU in queueAllocated is overuse, the GPU  must be overuse after podRequest + queueAllocated. So it is add besed on the  resourcelist of podRequest

Reply:
Got it. The logic is a little trick.
AddbaseLeft function
**input:** 
r  cpu 1
rr cpu1 gpu1
**output:**
r  cpu: 2

Add function
**input:** 
r: cpu 1
rr: cpu1 gpu1
**output:** 
r  cpu: 2, gpu: 1